Sources/StreamChat/Repositories/SyncRepository.swift

```diff
@@ -134,6 +137,12 @@ class SyncRepository {
                 )
             }
         operations.append(contentsOf: refetchChannelListQueryOperations)
+        let channelListRegistry = ChannelListRegistry()
+        let notification = Notification(name: SyncRepository.syncRepositoryChannelListQueryRegistrationNotification, object: channelListRegistry)
+        NotificationCenter.default.post(notification)
+        operations.append(contentsOf: channelListRegistry.registeredChannelListQueries.map { query in
+            RefetchChannelListQueryOperation(query: query, channelListUpdater: channelListUpdater, context: context)
+        })
```

Comment:
hm, can you elaborate a bit more on the issue? What do you mean by "no receive(on:) when using Combine"?

Reply:
I was thinking about this flow:

SyncRepository posts a notification on a background thread:
```swift
NotificationCenter.default.post(notification)
```
and someone is using such code for receiving the notification
```swift
… = NotificationCenter.default
        .publisher(for: SyncRepository.syncRepositoryChannelListQueryRegistrationNotification)
        .compactMap { $0.object as? SyncRepository.ChannelListRegistry }
        .receive(on: DispatchQueue.main)
        .sink(receiveValue: { [query] registry in registry.register(query: query) })
```
Then, when `post(notification)` has returned, the `sink` has not run yet because it was scheduled to run on a different thread. Meaning the query has not been added to the registry class and when SyncRepository accesses registered queries, it would miss that query (timing issue). As long as observers run observing code synchronously, it would be OK and query is added to the query before `post()` returns. 

But since it is a internal implementation then this downside should be OK.

Reply:
Alright, thanks for the explanation. Yes, probably better to use our wrapper for consistency.